src/LiveQuery/ParseWebSocketServer.js

```diff
@@ -14,17 +14,23 @@ export class ParseWebSocketServer {
       logger.info('Parse LiveQuery Server started running');
     };
     wss.onConnection = ws => {
+      ws.isAlive = true;
+      ws.on('pong', () => {
+        this.isAlive = true;
+      });
       ws.on('error', error => {
         logger.error(error.message);
         logger.error(inspect(ws, false));
       });
       onConnect(new ParseWebSocket(ws));
       // Send ping to client periodically
       const pingIntervalId = setInterval(() => {
-        if (ws.readyState == ws.OPEN) {
+        if (ws.isAlive) {
           ws.ping();
+          ws.isAlive = false;
```

Comment:
Could you change this logic so that either `ws.isAlive` only becomes `false` when it is not considered alive anymore (i.e. when `pong` times out), or rename the variable. Because semantically it is incorrect to say it is "not alive" while we are still trying to determine whether it is alive.

Reply:
If I want to change the variable name, what's your idea about "gotPong"?

Reply:
How about `waitingForPong`? That is the state that this var tries to describe I think.

Reply:
Thanks for the good suggestion, I changed the name.